core/retries/src/main/java/software/amazon/awssdk/retries/internal/BaseRetryStrategy.java

```diff
@@ -66,6 +71,8 @@ public abstract class BaseRetryStrategy implements RetryStrategy {
         this.treatAsThrottling = Validate.paramNotNull(builder.treatAsThrottling, "treatAsThrottling");
         this.exceptionCost = Validate.paramNotNull(builder.exceptionCost, "exceptionCost");
         this.tokenBucketStore = Validate.paramNotNull(builder.tokenBucketStore, "tokenBucketStore");
+        this.defaultsAdded = Validate.paramNotNull(builder.defaultsAdded, "defaultsAdded");
```

Comment:
You need to copy the set here instead of using the same as the builder, or, better yet, copy and pass it trough `Collections.unmodifiableSet`. Otherwise you end up sharing state that you shouldn't. For instance:

```java
RetryStategy.Builder builder = standardRetryStrategy().toBuilder();
addSomeFooDefaults(builder);
RetryStategy strategyA = builder.build();
addSomeBarDefaults(builder);
RetryStategy strategyB = builder.build();
```
After this, both, `strategyA` and `strategyB` think they have both, `Foo` and `Bar` defaults whereas only `strategyA` does.